File: 3375-minimum-operations-to-make-array-values-equal-to-k/3375-minimum-operations-to-make-array-values-equal-to-k.java
class Solution {
    public int minOperations(int[] nums, int k) {
        Set<Integer> st = new HashSet<>();
        int m = 101;
        for (int x : nums) {
            m = Math.min(m, x);
            st.add(x);
        }
        if (m < k) return -1;
        return st.size() + (m == k ? -1 : 0);
    }
}

File: 3375-minimum-operations-to-make-array-values-equal-to-k/3375-minimum-operations-to-make-array-values-equal-to-k.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        st = set(nums)
        m = min(nums)
        if m < k:
            return -1
        if m == k:
            return len(st) - 1
        else:
            return len(st)